gcc/rust/backend/rust-compile-fnparam.cc

```diff
@@ -117,5 +120,23 @@ CompileSelfParam::compile (Context *ctx, tree fndecl, HIR::SelfParam &self,
 						  locus);
 }
 
+// FIXME: I'm not sure if match_scrutinee fits as a name since it's used in
+// function parameters as well. Same for CompilePatternBindings'
+// match_scrutinee_expr
+tree
+CompileFnParam::create_tmp_match_scrutinee (tree decl_type)
+{
+  // generate the anon param
+  tree tmp_ident = create_tmp_var_name ("RSTPRM");
+  std::string cpp_str_identifier = std::string (IDENTIFIER_POINTER (tmp_ident));
+
+  decl_type = ctx->get_backend ()->immutable_type (decl_type);
+  compiled_param
+    = ctx->get_backend ()->parameter_variable (fndecl, cpp_str_identifier,
```

Comment:
That likely wont work for match expressions because we dont need parameters

Reply:
If I understand correctly, this part will never be reached by a `match` expression. It's only called from a `CompileFnParam::visit` which in turn is only called in function compilation and intrinsic functions. However, the naming for the function itself might be misleading since it has nothing to do with `match`, it was a misunderstanding on my part, I apologize.

Also, I think `StructPattern` and `TupleStructPattern` are not supported in match yet. We have no tests for them (only enums) and when I tried a test, it resulted in an ICE in the type-checking stage.